cmd/thanos/sidecar.go

```diff
@@ -125,6 +139,31 @@ func runSidecar(
 
 		ctx, cancel := context.WithCancel(context.Background())
 		g.Add(func() error {
+			// Only check Prometheus's flags when upload is enabled.
+			if uploads {
+				// Retry infinitely until we get Prometheus version.
```

Comment:
Maybe we should give it like X tries and then fail? X would be configurable parameter. 
As I don't think running infinetely is a good idea.

Reply:
The `/version` endpoint is currently protected by the readiness test in prometheus so it will only return when the WAL has been restored from disk, I think infinitely checking here is ok, as we'll never find "the right" value to timeout.

Reply:
Yea. We cannot start sidecar without working Prometheus anyway, (we need external labels and remote read), so it's fine. We already retry infinitely on external labels